bin/disco_elasticsearch.py

```diff
@@ -0,0 +1,110 @@
+    parser_create.set_defaults(mode="create")
+    parser_create.add_argument("--name", dest="name", type=str, action="append",
+                               help="Name of the ElasticSearch domain")
+
+    parser_update = subparsers.add_parser("update",
+                                          help="Update an ElasticSearch domain. If no options are provided, "
+                                          "default behavior is to update all ElasticSearch domains found in "
+                                          "the config.")
+    parser_update.set_defaults(mode="update")
+    parser_update.add_argument("--name", dest="name", type=str, action="append",
+                               help="Name of the ElasticSearch domain")
+
+    parser_delete = subparsers.add_parser("delete",
+                                          help="Delete an ElasticSearch domain. If no options are provided, "
+                                          "default behavior is to delete all ElasticSearch domains found in "
+                                          "the config.")
+    parser_delete.set_defaults(mode="delete")
+    parser_delete.add_argument("--name", dest="name", type=str, action="append",
+                               help="Name of the ElasticSearch domain")
+    parser_delete.add_argument("--all", dest="delete_all", action='store_const', default=False, const=True,
+                               help="Delete *all* ElasticSearch domains")
+
+    return parser
+
+
+def run():
+    """Parses command line and dispatches the commands"""
+    parser = get_parser()
+    args = parser.parse_args()
+    configure_logging(args.debug)
+    env = args.env
+    disco_es = DiscoElasticsearch(env)
+
+    if args.mode == "list":
+        entries = disco_es.list(include_endpoint=args.endpoint)
+        headers = ["Elastic Search Domain Name", "Internal Name", "Route 53 Endpoint"]
+        format_line = u"{0:<28} {1:<15} {2:<35}"
+        if args.endpoint:
+            format_line += u" {3:<80}"
+            headers.append("Elastic Search Endpoint")
+        print(format_line.format(*headers), file=sys.stderr)
+        for entry in entries:
+            values = [entry["elasticsearch_domain_name"], entry["internal_name"], entry["route_53_endpoint"]]
+            if args.endpoint:
+                values.append(entry["elasticsearch_endpoint"] or u"-")
+            print(format_line.format(*values))
+
+    elif args.mode == "create":
+        if args.name:
+            for name in args.name:
+                disco_es.create(name)
+        else:
+            disco_es.create()
+
+    elif args.mode == "update":
+        if args.name:
+            for name in args.name:
+                disco_es.update(name)
+        else:
+            disco_es.update()
+
+    elif args.mode == "delete":
+        print("Deleting an ElasticSearch domain destroys all automated snapshots of its data. Be careful!")
+        if args.name:
```

Comment:
This sanity check might complicate automated invocation. One work around is checking if interactive shell
http://stackoverflow.com/questions/6108330/checking-for-interactive-shell-in-a-python-script